app/graph/page.tsx

```diff
@@ -76,7 +87,10 @@ export default function Page() {
         const newGraph = Graph.create(state.graphName, json.result)
         setGraph(newGraph)
         setMetaData(json.result.metadata)
-        setShowGraph((!!json.result.data && json.result.data.length > 0))
+        console.log(json.result.data);
+            
+            setShowGraph(!!json.result.data && typeof Object.values(json.result.data[0])[0] === "object")
+            setShowData((!!json.result.data && json.result.data.length > 0))
```

Comment:
```suggestion
            if(!!json.result.data){
                setShowGraph(typeof Object.values(json.result.data[0])[0] === "object")
                setShowData(json.result.data.length > 0)
            }
```

Reply:
This check won't work if you have a mix of scalar and entities in the result set.
You should check after the extract data if a Graph was created.